gdal/frmts/dimap/dimapdataset.cpp

```diff
@@ -1333,6 +1334,21 @@ int DIMAPDataset::ReadImageInformation2()
         }
     }
 
+/* -------------------------------------------------------------------- */
+/*      Expose Overviews if available                                   */
+/* -------------------------------------------------------------------- */
+    auto poSrcBandFirstImage = poImageDS->GetRasterBand(1);
+    const int nSrcOverviews = std::min(30, poSrcBandFirstImage->GetOverviewCount());
+    if(nSrcOverviews>0) {
+        std::unique_ptr<int[]> ovrLevels(new int[nSrcOverviews]);
+        int iLvl=2;
+        for(int i=0; i<nSrcOverviews; i++) {
+            ovrLevels[i]=iLvl;
+            iLvl*=2;
```

Comment:
Actually for nSrcOverviews == 30, we would overvflow int at the last iteration...
```suggestion
        int iLvl=1;
        for(int i=0; i<nSrcOverviews; i++) {
            iLvl*=2;
            ovrLevels[i]=iLvl;
```